src/ppk2_api/ppk2_api.py

```diff
@@ -219,13 +219,13 @@ def list_devices():
             devices = [
                 (port.device, port.serial_number[:8])
                 for port in ports
-                if port.description.startswith("nRF Connect USB CDC ACM")
+                if port.description.startswith("nRF Connect USB CDC ACM") and port.location.endswith("1")
```

Comment:
seems that location is same for both ports in macOS. But seems that first occurrence works there.

Reply:
I unfortunately don't have a mac to test with, but "first occurrence"; can you elaborate? The location should be derived from the USB hub tree, and both having the same location doesn't make sense to me as there has to be two logical endpoints. on the device.

Reply:
Since I haven't heard anything in a while, I expect this to be working by chance at least on MacOS as well :)